### ISSUE TITLE
Error during installation: The filename or extension is too long:

### ISSUE THREAD

**Issue by jsteyn** [carpentriesoffline/offlinedatasci#140]
I thought this was resolved but I got this error again. I first tried pip and then pipx. Here is the output from pipx:

```
Microsoft Windows [Version 10.0.19045.4780]
(c) Microsoft Corporation. All rights reserved.

C:\Users\janne>pipx install offlinedatasci
Fatal error from pip prevented installation. Full pip output in file:
    C:\Users\janne\pipx\logs\cmd_2024-09-02_22.48.04_pip_errors.log

pip seemed to fail to build package:
    certifi>=2017.4.17

Some possibly relevant errors from pip install:
    ERROR: Could not install packages due to an OSError: [WinError 206] The filename or extension is too long: 'C:\\Users\\janne\\pipx\\venvs\\offlinedatasci\\Lib\\site-packages\\build\\lib\\build\\lib\\build\\lib\\build\\lib\\build\\lib\\build\\lib\\build\\lib\\build\\lib\\build\\lib\\build\\lib\\build\\lib\\build\\lib\\build\\lib\\build\\lib\\build\\lib\\build\\lib\\build\\lib\\build\\lib\\build\\lib'

Error installing offlinedatasci.

C:\Users\janne>
```

==============================
The content of the mentioned log file:

```
C:\Users\janne>type C:\Users\janne\pipx\logs\cmd_2024-09-02_22.48.04_pip_errors.log
PIP STDOUT
----------
Collecting offlinedatasci
  Using cached offlinedatasci-0.6.0-py3-none-any.whl.metadata (4.8 kB)
Collecting airium (from offlinedatasci)
  Using cached airium-0.2.6-py3-none-any.whl.metadata (15 kB)
Collecting beautifulsoup4 (from offlinedatasci)
  Using cached beautifulsoup4-4.12.3-py3-none-any.whl.metadata (3.8 kB)
Collecting importlib-resources (from offlinedatasci)
  Using cached importlib_resources-6.4.4-py3-none-any.whl.metadata (4.0 kB)
Collecting lxml (from offlinedatasci)
  Using cached lxml-5.3.0-cp312-cp312-win_amd64.whl.metadata (3.9 kB)
Collecting python-pypi-mirror (from offlinedatasci)
  Using cached python_pypi_mirror-5.2.1-py3-none-any.whl.metadata (3.4 kB)
Collecting requests (from offlinedatasci)
  Using cached requests-2.32.3-py3-none-any.whl.metadata (4.6 kB)
Collecting setuptools (from offlinedatasci)
  Using cached setuptools-74.1.0-py3-none-any.whl.metadata (6.7 kB)
Collecting soupsieve>1.2 (from beautifulsoup4->offlinedatasci)
  Using cached soupsieve-2.6-py3-none-any.whl.metadata (4.6 kB)
Collecting charset-normalizer<4,>=2 (from requests->offlinedatasci)
  Using cached charset_normalizer-3.3.2-cp312-cp312-win_amd64.whl.metadata (34 kB)
Collecting idna<4,>=2.5 (from requests->offlinedatasci)
  Using cached idna-3.8-py3-none-any.whl.metadata (9.9 kB)
Collecting urllib3<3,>=1.21.1 (from requests->offlinedatasci)
  Using cached urllib3-2.2.2-py3-none-any.whl.metadata (6.4 kB)
Collecting certifi>=2017.4.17 (from requests->offlinedatasci)
  Using cached certifi-2024.8.30-py3-none-any.whl.metadata (2.2 kB)
Using cached offlinedatasci-0.6.0-py3-none-any.whl (206 kB)
Using cached airium-0.2.6-py3-none-any.whl (13 kB)
Using cached beautifulsoup4-4.12.3-py3-none-any.whl (147 kB)
Using cached importlib_resources-6.4.4-py3-none-any.whl (35 kB)
Using cached lxml-5.3.0-cp312-cp312-win_amd64.whl (3.8 MB)
Using cached python_pypi_mirror-5.2.1-py3-none-any.whl (10 kB)
Using cached requests-2.32.3-py3-none-any.whl (64 kB)
Using cached setuptools-74.1.0-py3-none-any.whl (1.3 MB)
Using cached certifi-2024.8.30-py3-none-any.whl (167 kB)
Using cached charset_normalizer-3.3.2-cp312-cp312-win_amd64.whl (100 kB)
Using cached idna-3.8-py3-none-any.whl (66 kB)
Using cached soupsieve-2.6-py3-none-any.whl (36 kB)
Using cached urllib3-2.2.2-py3-none-any.whl (121 kB)
Installing collected packages: airium, urllib3, soupsieve, setuptools, python-pypi-mirror, lxml, importlib-resources, idna, charset-normalizer, certifi, requests, beautifulsoup4, offlinedatasci

PIP STDERR
----------
ERROR: Could not install packages due to an OSError: [WinError 206] The filename or extension is too long: 'C:\\Users\\janne\\pipx\\venvs\\offlinedatasci\\Lib\\site-packages\\build\\lib\\build\\lib\\build\\lib\\build\\lib\\build\\lib\\build\\lib\\build\\lib\\build\\lib\\build\\lib\\build\\lib\\build\\lib\\build\\lib\\build\\lib\\build\\lib\\build\\lib\\build\\lib\\build\\lib\\build\\lib\\build\\lib'


C:\Users\janne>
```

**Comment by ethanwhite** [comment]
Thanks for the report. We had fixed the original issue in 9109e38, but I can confirm that a fresh miniconda install on Windows produces this error.

**Comment by ethanwhite** [comment]
Just to make this extra confusing, installing using pip based on the repo works fine:

```
pip install git+https://github.com/carpentriesoffline/offlinedatasci.git
```

I've checked the tarball on pypi and the line that was causing the issue isn't there (so it wasn't a fumbled release build).

**Comment by ethanwhite** [comment]
Installing from the tarball on pip also works:

```
pip install --no-binary offlinedatasci offlinedatasci
```

Best guess now is that I accidentally uploaded an old wheel...

**Comment by ethanwhite** [comment]
Nope, the code changes are in there, but so is the recursive directory structure. It looks like a fresh wheel build is going to fix this so I'm going to try a quick patch release.

**Comment by ethanwhite** [comment]
Very weird, but should be fixed with the release of `0.6.1`.

@jsteyn can you confirm that things are working for you now?
